src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -220,6 +220,10 @@ private void applyCleanedIndices(final ClusterChangedEvent event) {
         // handle closed indices, since they are not allocated on a node once they are closed
         // so applyDeletedIndices might not take them into account
         for (IndexService indexService : indicesService) {
+            if (indexService == null) {
```

Comment:
checking here for null is not benefetial I think, it can't be null...., same applies to other null checks while iterating